tests/unit/backend/recipe/algorithm/data/test_Leftovers.py

```diff
@@ -0,0 +1,65 @@
+import tempfile
+
+import numpy as np
+from mantid.api import mtd
+from mantid.simpleapi import *
+from snapred.backend.recipe.algorithm.data.ReheatLeftovers import ReheatLeftovers
+from snapred.backend.recipe.algorithm.data.WrapLeftovers import WrapLeftovers
+
+
+def compareRaggedWorkspaces(ws1, ws2):
+    assert ws1.getNumberHistograms() == ws2.getNumberHistograms()
+
+    assert ws1.isRaggedWorkspace()
+    assert ws2.isRaggedWorkspace()
+
+    for i in range(ws1.getNumberHistograms()):
+        np.testing.assert_allclose(ws1.readX(i), ws2.readX(i))
+        np.testing.assert_allclose(ws1.readY(i), ws2.readY(i))
+        np.testing.assert_allclose(ws1.readE(i), ws2.readE(i))
```

Comment:
Just before this, could you add an assert that
``` python
assert ws1.getNumberHistograms() == ws2.getNumberHistograms()
```
?

Reply:
Also an assert that spectrum numbers are equal at each workspace index?

Reply:
I think it's
```
assert ws1.getSepctrumNumbers() == ws2.getSpectrumNumbers()
```

Reply:
Probably something like
``` python
assert ws1.getNumberHistograms() == ws2.getNumberHistograms()
assert ws1.getSpectrumNumbers() == ws2.getSpectrumNumbers()
for i in range(ws1.getNumberHistograms():
    spec1 = ws1.getSpectrum(i)
    spec2 = ws2.getSpectrum(i)
    assert spec1.getDetectorIDs() == spec2.getDetectorIDs()
    assert spec1.getSpectrumNo() == spec2.getSpectrumNo()
    np.testing.assert_allclose(ws1.readX(i), ws2.readX(i)
    etc.
```

Reply:
added roughly the equivalent